src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/Handler/RefundItemsTransactionHandler.php

```diff
@@ -0,0 +1,218 @@
+ */
+
+namespace SprykerEco\Zed\Braintree\Business\Payment\Transaction\Handler;
+
+use ArrayObject;
+use Generated\Shared\Transfer\ExpenseTransfer;
+use Generated\Shared\Transfer\ItemTransfer;
+use Generated\Shared\Transfer\RefundTransfer;
+use Generated\Shared\Transfer\TransactionMetaTransfer;
+use Orm\Zed\Sales\Persistence\SpySalesOrder;
+use Spryker\Shared\Shipment\ShipmentConstants;
+use SprykerEco\Zed\Braintree\Business\Payment\Transaction\MetaVisitor\TransactionMetaVisitorInterface;
+use SprykerEco\Zed\Braintree\Business\Payment\Transaction\TransactionInterface;
+use SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToRefundFacadeInterface;
+use SprykerEco\Zed\Braintree\Persistence\BraintreeRepositoryInterface;
+
+class RefundItemsTransactionHandler extends AbstractTransactionHandler implements RefundItemsTransactionHandlerInterface
+{
+    protected const KEY_AMOUNT = 'amount';
+    protected const KEY_PAYMENT_ID = 'payment_id';
+
+    /**
+     * @var \SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToRefundFacadeInterface
+     */
+    protected $refundFacade;
+
+    /**
+     * @var \SprykerEco\Zed\Braintree\Persistence\BraintreeRepositoryInterface
+     */
+    protected $braintreeRepository;
+
+    /**
+     * @param \SprykerEco\Zed\Braintree\Business\Payment\Transaction\TransactionInterface $transaction
+     * @param \SprykerEco\Zed\Braintree\Business\Payment\Transaction\MetaVisitor\TransactionMetaVisitorInterface $transactionMetaVisitor
+     * @param \SprykerEco\Zed\Braintree\Dependency\Facade\BraintreeToRefundFacadeInterface $refundFacade
+     * @param \SprykerEco\Zed\Braintree\Persistence\BraintreeRepositoryInterface $braintreeRepository
+     */
+    public function __construct(
+        TransactionInterface $transaction,
+        TransactionMetaVisitorInterface $transactionMetaVisitor,
+        BraintreeToRefundFacadeInterface $refundFacade,
+        BraintreeRepositoryInterface $braintreeRepository
+    ) {
+        parent::__construct($transaction, $transactionMetaVisitor);
+
+        $this->refundFacade = $refundFacade;
+        $this->braintreeRepository = $braintreeRepository;
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $salesOrderItems
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrder $salesOrderEntity
+     *
+     * @return void
+     */
+    public function refund(array $salesOrderItems, SpySalesOrder $salesOrderEntity): void
+    {
+        $refundTransfer = $this->getRefund($salesOrderItems, $salesOrderEntity);
+        $transactionMetaTransfer = $this->getTransactionMetaTransfer($salesOrderItems, $salesOrderEntity, $refundTransfer);
+
+        $orderItemsGroupedByTransaction = $this->getOrderItemsGroupedByTransaction($transactionMetaTransfer);
+        $shipmentExpense = $this->getShipmentExpenseTransfer($refundTransfer);
+
+        foreach ($orderItemsGroupedByTransaction as $transactionId => $data) {
```

Comment:
would be nice to have this loop in the separate method with descriptive name

Reply:
Done